### DISCUSSION TITLE
Invalid Habitat code

### DISCUSSION THREAD

**Discussion by bs17smh** [RangeShifter/RangeshiftR-tutorials#28] [Q&A] [answered]
Hello, 

When running a simulation using a ascii raster file created from a UK landcover map and some manually drawn hedgerows I get the error 

"Run Simulation(s) with random seed ...
LandscapeFile headers OK: E:/OneDrive - University of Leeds/PhD/chap3/Rshift/Landscape_1/Mammal/simad/Inputs/habitat_1.asc
PatchFile headers OK: E:/OneDrive - University of Leeds/PhD/chap3/Rshift/Landscape_1/Mammal/simad/Inputs/patch_1.asc
Found invalid habitat code."

This raster works in the GUI version just not in the R version and when I list the habitat codes:

unique(lcm_reclass)
  lcm2015gb25m_1
1              1
2              2
3              5
4              6
5             20

they are all just integers and should be fine? 
I'm just not sure where to look to start fixing this

cheers for any help,
Sam

**Comment by annekathleenmalchow** [comment, marked as answer]
Hi Sam,
the cause for this is one of the few differences that remained between the GUI and the batch version & RangeShiftR (which is based on the batch version). 
In the GUI the habitat codes can take any positive integer values, like 1,5,7,... but in the batch version & in RangeShiftR, the habitat codes must be sequential, like 1,2,3,... . If you convert the values of your habitat types, it should work.
Hope this helps!
Cheers,
Anne

**Reply by bs17smh** [reply]
Brilliant, probably should've tried that before I posted:) So simple

best.
Sam
